search/includes/functions/helpers.php

```diff
@@ -0,0 +1,18 @@
+<?php
+// phpcs:disable WordPress.PHP.YodaConditions.NotYoda
+namespace Automattic\VIP\Search;
+
+// source: Laravel Framework
+// https://github.com/laravel/framework/blob/8.x/src/Illuminate/Support/Str.php
+
+function str_starts_with( $haystack, $needle ) {
+	return (string) $needle !== '' && strncmp( $haystack, $needle, strlen( $needle ) ) === 0;
+}
+
+function str_ends_with( $haystack, $needle ) {
+	return $needle !== '' && substr( $haystack, -strlen( $needle ) ) === (string) $needle;
```

Comment:
Should we cast `$needle` to `string` here like we do this in `str_starts_with()`?

Or, what do you think if we add types to function arguments, like this:

```php
function str_starts_with( string $haystack, string $needle ): bool
```

Then we won't need explicit type casts…